File: app/src/main/java/com/khmelenko/lab/varis/network/request/AuthorizationRequest.java
package com.khmelenko.lab.varis.network.request;

import com.google.gson.annotations.SerializedName;

import java.util.ArrayList;
import java.util.List;

/**
 * Authorization request
 *
 * @author Dmytro Khmelenko
 */
public class AuthorizationRequest {

    @SerializedName("scopes")
    private final List<String> mScopes;

    @SerializedName("note")
    private final String mNote;

    @SerializedName("note_url")
    private String mNoteUrl;

    @SerializedName("client_id")
    private String mClientId;

    @SerializedName("client_secret")
    private String mClientSecret;

    @SerializedName("fingerprint")
    private String mFingerprint;

    public AuthorizationRequest(List<String> scopes, String note) {
        mScopes = new ArrayList<>(scopes);
        mNote = note;
    }

    public List<String> getScopes() {
        return mScopes;
    }

    public String getNote() {
        return mNote;
    }

    public String getNoteUrl() {
        return mNoteUrl;
    }

    public void setNoteUrl(String noteUrl) {
        mNoteUrl = noteUrl;
    }

    public String getClientId() {
        return mClientId;
    }

    public void setClientId(String clientId) {
        mClientId = clientId;
    }

    public String getClientSecret() {
        return mClientSecret;
    }

    public void setClientSecret(String clientSecret) {
        mClientSecret = clientSecret;
    }

    public String getFingerprint() {
        return mFingerprint;
    }

    public void setFingerprint(String fingerprint) {
        mFingerprint = fingerprint;
    }
}

File: app-v3/src/main/java/com/khmelenko/lab/varis/network/request/AuthorizationRequest.kt
package com.khmelenko.lab.varis.network.request

import com.google.gson.annotations.SerializedName

/**
 * Authorization request
 *
 * @author Dmytro Khmelenko
 */
data class AuthorizationRequest(
        @SerializedName("scopes") val scopes: List<String>,
        @SerializedName("note") val note: String,
        @SerializedName("note_url") val noteUrl: String? = null,
        @SerializedName("client_id") val clientId: String? = null,
        @SerializedName("client_secret") val clientSecret: String? = null,
        @SerializedName("fingerprint") val fingerprint: String? = null
)
